app/router/job.py

```diff
@@ -48,27 +48,32 @@ def get_jobs(
         log(log.INFO, "Job filtered")
     else:
         profession_ids: list[int] = [profession.id for profession in user.professions]
-        cities_name: list[str] = [location.name_en for location in user.locations]
+        cities_names: list[str] = [location.name_en for location in user.locations]
         if profession_ids:
-            query = query.where(m.Job.profession_id.in_(profession_ids))
+            filter_conditions = []
+            for prof_id in profession_ids:
+                filter_conditions.append(m.Job.profession_id == prof_id)
+            query = query.filter(or_(*filter_conditions))
+
             log(
                 log.INFO,
                 "Job filtered by profession ids [%s] user interests",
                 profession_ids,
             )
-        if cities_name:
-            query = query.where(
-                func.lower(m.Job.city).in_([city.lower() for city in cities_name])
-            )
+        if cities_names:
+            filter_conditions = []
+            for location in cities_names:
+                filter_conditions.append(m.Job.city == location)

```

Comment:
try to use comparison patterns ilike or contains 
we want to handle some small typos